### ISSUE TITLE
`use_debug` lint should not trigger on `debug!()` log macro calls

### ISSUE THREAD

**Issue by hcpl** [rust-lang-nursery/rust-clippy#2132]
Example use-case:
```rust
#![cfg_attr(feature = "cargo-clippy", warn(use_debug))]

#[macro_use]
extern crate log;

pub fn euclid_steps(mut x: u64, mut y: u64) -> Vec<u64> {
    debug!("x = {}, y = {}", x, y);

    let mut steps = vec![];

    loop {
        let tmp = x % y;

        if tmp == 0 {
            debug!("Steps performed in Euclidean algorithm: {:?}", steps);
            return steps;
        }

        steps.push(tmp);
        x = y;
        y = tmp;
    }
}
```

**Comment by oli-obk** [comment]
should be very easy with the new system. basically we only want this lint on explicit calls to `print!`, `println!`, `write!` and `writeln!`

**Comment by sanmai-NL** [comment]
What about `format!` and then reusing the resulting string in some inappropriate way, e.g. printing the string later on?

**Comment by oli-obk** [comment]
that, too :D good catch

**Comment by sanmai-NL** [comment]
I’m trying to say, when the programmer uses `format!`, it’s unclear what happens with the output and also whether that’s inappropriate.

**Comment by oli-obk** [comment]
oh... seems to me that this would be a case for "moderate sprinkling of allow"?

**Comment by sanmai-NL** [comment]
So deny on `format!` by default? Seems fine to me.
